### PR TITLE
Geo: add validator that only checks altitude

### PR DESCRIPTION
**Pull request by imotov**
By default, we don't check ranges while indexing geo_shapes. As a
result, it is possible to index geoshapes that contain contain
coordinates outside of -90 +90 and -180 +180 ranges. Such geoshapes
will currently break SQL and ML retrieval mechanism. This commit removes
 these restriction from the validator is used in SQL and ML retrieval.

### CODE REVIEW

File: libs/geo/src/main/java/org/elasticsearch/geo/utils/StandardValidator.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import org.elasticsearch.geo.geometry.LinearRing;
+import org.elasticsearch.geo.geometry.MultiLine;
+import org.elasticsearch.geo.geometry.MultiPoint;
+import org.elasticsearch.geo.geometry.MultiPolygon;
+import org.elasticsearch.geo.geometry.Point;
+import org.elasticsearch.geo.geometry.Polygon;
+import org.elasticsearch.geo.geometry.Rectangle;
+
+/**
+ * Validator that only checks that checks that altitude only shows up if ignoreZValue is set to true.
```
Comment (talevy) [inline libs/geo/src/main/java/org/elasticsearch/geo/utils/StandardValidator.java]: repeat `checks`?

**Review by talevy — COMMENTED** [review]
small nit in the javadoc, mostly LGTM

I have one question, which might be unrelated to this PR, but more of a general 
testing consideration for GeometryVisitors. It feels like we should be testing
all calls to `visit` for each of the different shapes. is this being done somewhere
that I am not aware of?

**Review by talevy — APPROVED** [review]
thanks!

**Comment by imotov** [comment]
@elasticmachine test this please

**Pull request merged**